quixstreams/dataframe/dataframe.py

```diff
@@ -1467,6 +1502,39 @@ def _sink_callback(
         # uses apply without returning to make this operation terminal
         self.apply(_sink_callback, metadata=True)
 
+    def concat(self, other: Self) -> Self:
+        """
+        Concatenate two StreamingDataFrames together and return a new one.
+        The transformations applied on this new StreamingDataFrame will update data
+        from both origins.
+
+        Use it to concatenate dataframes belonging to different topics as well as to merge the branches
+        of the same original dataframe.
+
+        If concatenated dataframes belong to different topics, the stateful operations
+        on the new dataframe will create different state stores
+        unrelated to the original dataframes and topics.
+        The same is true for the repartition topics created by `.group_by()`.
+
+        :param other: other StreamingDataFrame
+        :return: a new StreamingDataFrame
+
+        """
+        merged_stream = self.stream.merge(other.stream)
+        topics = set(self.topics) | set(other.topics)
+        return self.__dataframe_clone__(*topics, stream=merged_stream)
```

Comment:
Technically, since the `StreamingDataFrame.__init__` is converting `topics` to set, here we could skip it. This will work out of the box:
```suggestion
        stream = self.stream.merge(other.stream)
        return self.__dataframe_clone__(*self.topics, *other.topics, stream=stream)
```

Reply:
Yep, I like that